### PR TITLE
Scheduling generator

### LINKED ISSUE #14: Scheduling generator
Create instances as in chapter 8 of this [paper](https://www.researchgate.net/profile/John_Hooker/publication/221633529_Planning_and_Scheduling_to_Minimize_Tardiness/links/0deec5321d2c39536b000000/Planning-and-Scheduling-to-Minimize-Tardiness.pdf) and chapter 4.3 of this [paper](https://opus4.kobv.de/opus4-zib/files/1767/ZR-13-05.pdf). The later is based on the former.

### PR DESCRIPTION
**Pull request by mmghannam**
Closes #14

### CODE REVIEW

**Comment by CharJon** [comment]
Please see update in #14.

File: mpgen/hooker.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            vars += [
+                c[i, j] * x[i, j, t_prime]
+                for t_prime in range(t - p[i, j] + 1, t + 1)
+                if (i, j, t_prime) in x
+            ]
+        model.addCons(scip.quicksum(vars) <= C[i])
+
+    # constraint (d)
+    for j, t in itertools.product(range(number_of_tasks), range(time_steps)):
+        if t < r[j] or t > time_steps - p[i, j]:
```
Comment (CharJon) [inline mpgen/hooker.py:113]: Is the 'i' in this line correct?

Comment (mmghannam) [reply mpgen/hooker.py:113]: yes it is, but I forget to add the i in the loop. This check should be made on all assignment variables Xijk

Comment (mmghannam) [reply mpgen/hooker.py:113]: Should be correct now I think 😄

**Comment by CharJon** [comment]
In cases where the number of constraints is not fixed we could still give lower and upper bound in the assertions.

File: mpgen/scheduling.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    for j, k in itertools.product(range(number_of_facilities), range(number_of_tasks)):
+        var = model.addVar(lb=0, ub=1, obj=c[j, k], name=f"x_{j}_{k}", vtype="B")
+        x[j, k] = var
+
+    # y vars
+    y = {}
+    for j, k, t in itertools.product(
+        range(number_of_facilities), range(number_of_tasks), range(time_steps)
+    ):
+        var = model.addVar(lb=0, ub=1, obj=0, name=f"y_{j}_{k}_{t}", vtype="B")
```
[the remark this answers is not in the document]
Comment (mmghannam) [reply mpgen/scheduling.py]: Fixed.

**Pull request merged**